bin/node/src/chain_spec.rs

```diff
@@ -38,30 +47,85 @@ where
     AccountPublic::from(get_from_seed::<TPublic>(seed)).into_account()
 }
 
-fn read_keys(n_members: usize) -> (Vec<AuraId>, Vec<AlephId>) {
-    let auth_keys = std::fs::read_to_string(KEY_PATH).expect("keys were not generated");
+#[derive(Clone)]
+struct AuthorityKeys {
+    account_id: AccountId,
+    aura_key: AuraId,
+    aleph_key: AlephId,
+}
+
+fn read_keys(n_members: usize) -> Vec<AuthorityKeys> {
     let auth_keys: HashMap<u32, Vec<[u8; 32]>> =
-        serde_json::from_str(&auth_keys).expect("should contain list of keys");
+        if let Ok(auth_keys) = std::fs::read_to_string(KEY_PATH) {
+            serde_json::from_str(&auth_keys).expect("should contain list of keys")
+        } else {
+            return Default::default();
+        };
 
-    let aura_keys: Vec<_> = auth_keys
+    let aura_keys = auth_keys
         .get(&key_types::AURA.into())
         .unwrap()
         .iter()
-        .take(n_members)
         .copied()
-        .map(|bytes| AuraId::from(sr25519::Public::from_raw(bytes)))
-        .collect();
+        .map(|bytes| AuraId::from(sr25519::Public::from_raw(bytes)));
 
-    let aleph_keys: Vec<_> = auth_keys
+    let aleph_keys = auth_keys
         .get(&aleph_primitives::KEY_TYPE.into())
         .unwrap()
         .iter()
-        .take(n_members)
         .copied()
-        .map(|bytes| AlephId::from(ed25519::Public::from_raw(bytes)))
-        .collect();
+        .map(|bytes| AlephId::from(ed25519::Public::from_raw(bytes)));
 
-    (aura_keys, aleph_keys)
+    let account_ids = LOCAL_AUTHORITIES
+        .iter()
+        .map(get_account_id_from_seed::<sr25519::Public>);
+
+    aura_keys
+        .zip(aleph_keys)
+        .zip(account_ids)
+        .take(n_members)
+        .map(|((aura_key, aleph_key), account_id)| AuthorityKeys {
+            aura_key,
+            aleph_key,
+            account_id,
+        })
+        .collect()
+}
+
+// TODO: rename to AlephEnvVarParams
+#[derive(Clone, Copy)]
+struct EnvironmentVariables {
+    session_period: u32,
+    millisecs_per_block: u64,
+}
+
+impl EnvironmentVariables {
+    fn fetch() -> Result<Self, String> {
+        Ok(Self {
+            session_period: Self::fetch_var_or(SESSION_PERIOD_ENV_VAR, DEFAULT_SESSION_PERIOD)?,
```

Comment:
YMMV but if the defaults are only read here, there might be no point in storing them as const, just hardcode them here directly?